ntoskrnl/ps/process.c

```diff
@@ -1614,4 +1677,234 @@ NtOpenProcess(OUT PHANDLE ProcessHandle,
     /* Return status */
     return Status;
 }
+
+PEPROCESS NtpGetAdjacentProcess(const BOOLEAN is_backwards, const PEPROCESS current)
+{
+    return !is_backwards
+               ? PsGetNextProcess(current)
+               : PsGetPreviousProcess(current);
+}
+
+NTSTATUS
+NTAPI
+NtGetNextProcess(
+    _In_ HANDLE ProcessHandle,
+    _In_ ACCESS_MASK DesiredAccess,
+    _In_ ULONG HandleAttributes,
+    _In_ ULONG Flags,
+    _Out_ PHANDLE NewProcessHandle
+)
+{
+    const KPROCESSOR_MODE previousMode = KeGetPreviousMode();
+
+    /* Validate user attributes */
+    HandleAttributes = ObpValidateAttributes(HandleAttributes, previousMode);
+
+    /* Check if we came from user mode */
+    if (previousMode != KernelMode)
+    {
+        _SEH2_TRY
+        {
+            /* Probe process handle */
+            ProbeForWriteHandle(NewProcessHandle);
+
+            *NewProcessHandle = NULL;
+        }
+        _SEH2_EXCEPT(EXCEPTION_EXECUTE_HANDLER)
+        {
+            /* Return the exception code */
+            _SEH2_YIELD(return _SEH2_GetExceptionCode());
+        }
+        _SEH2_END;
+    }
+    else
+    {
+        *NewProcessHandle = NULL;
+    }
+
+    const BOOLEAN is_backwards = Flags & 1;
```

Comment:
The `1` should be a define in NDK rather than a magic value.

Reply:
Since this function is part of private NT API, there is no official flag name. Flag was found by bruteforcing accepted bit values.

Reply:
The idea behind NDK is to make the native API more accessible, so it's exactly where such flags go (if the constant  isn't publicly documented, you simply make up a descriptive name).
However even if the value were completely internal to the kernel, it's bad practice to use "magic numbers" in code, and would be better to add a define local to the C file.